src/components/common/ColonyFunding/ColonyFunding.tsx

```diff
@@ -1,149 +1,141 @@
-  // }, [domainChoices, formatMessage, selectedDomainId]);
+  const domainChoices = useMemo<
+    ComponentProps<typeof Select>['options']
+  >(() => {
+    if (!domains?.items) {
+      return [];
+    }
+    return [
+      {
+        value: COLONY_TOTAL_BALANCE_DOMAIN_ID.toString(),
+        label: { id: 'domain.all' },
+      },
+      ...(domains?.items ?? [])
+        .map((domain) => ({
+          label: domain?.name ?? '',
+          value: domain?.nativeId.toString() ?? '',
+        }))
+        .sort(
+          (first, second) =>
+            parseInt(first.value, 10) - parseInt(second.value, 10),
+        ),
+    ];
+  }, [domains]);
 
-  // if (
-  //   loading ||
-  //   (data?.colonyAddress &&
-  //     !data.processedColony &&
-  //     !((data.colonyAddress as any) instanceof Error))
-  // ) {
-  //   return (
-  //     <div className={styles.loadingWrapper}>
-  //       <LoadingTemplate loadingText={MSG.loadingText} />
-  //     </div>
-  //   );
-  // }
+  const selectedDomainLabel: string = useMemo(() => {
+    const { label = '' } =
+      domainChoices.find(
+        ({ value }) => value === selectedDomainId.toString(),
+      ) || {};
+    return typeof label === 'string' ? label : formatMessage(label);
+  }, [domainChoices, formatMessage, selectedDomainId]);
 
-  // if (!colonyName || error || !data?.processedColony) {
-  //   console.error(error);
-  //   return <NotFoundRoute />;
-  // }
-
-  // const { processedColony: colony } = data;
+  // eslint-disable-next-line react/no-unstable-nested-components
+  const Aside = () => (
+    <aside className={styles.aside}>
+      <ColonyFundingMenu />
+    </aside>
+  );
 
   return (
     <div className={styles.main}>
       <div className={styles.content}>
         <div>
+          {isMobile && <ColonyHomeInfo showNavigation isMobile />}
           <div className={styles.titleContainer}>
             <Heading
               text={MSG.title}
               textValues={{ selectedDomainLabel }}
               appearance={{ size: 'medium', theme: 'dark' }}
             />
-            {/* <Form
-              initialValues={{
-                selectDomain: COLONY_TOTAL_BALANCE_DOMAIN_ID.toString(),
-              }}
-              onSubmit={() => {}}
-            >
-              <Select
-                appearance={{
-                  alignOptions: 'right',
-                  width: 'content',
-                  theme: 'alt',
-                }}
-                elementOnly
-                label={MSG.labelSelectDomain}
-                name="selectDomain"
-                onChange={(value) => setSelectedDomainId(Number(value))}
-                options={domainChoices}
+            {isMobile ? (
+              <ColonyDomainSelector
+                filteredDomainId={selectedDomainId}
+                onDomainChange={setSelectedDomainId}
               />
-            </Form> */}
+            ) : (
+              <Form
```

Comment:
Again, good improvement, but not going to do it, since this potentially wasted time. See: https://github.com/JoinColony/colonyCDapp/pull/74#discussion_r1131145391